modules/private.php

```diff
@@ -0,0 +1,265 @@
+<?php
+/**
+ * Module Name: Private site
+ * Module Description: Make your site only visible to you and users you approve.
+ * Sort Order: 9
+ * First Introduced: ?
+ * Requires Connection: No
+ * Auto Activate: No
+ * Module Tags: Private
+ * Feature: Traffic
+ * Additional Search Queries: private, sandbox, launch, unlaunched
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Returns the private site template for private blogs
+ *
+ * @param object $wp Current WordPress environment instance (passed by reference).
+ */
+function privatize_blog( $wp ) {
+	global $pagenow, $current_user, $wpdb;
+
+	if ( '-1' !== get_option( 'blog_public' ) ) {
+		return;
+	}
+
+	if ( 'wp-login.php' === $pagenow ) {
+		return;
+	}
+
+	if ( defined( 'WP_CLI' ) && WP_CLI ) {
+		return;
+	}
+
+	// Serve robots.txt for private blogs.
+	if ( is_object( $wp ) && ! empty( $wp->query_vars['robots'] ) ) {
+		return;
+	}
+
+	if ( $current_user && ( is_super_admin() || is_private_blog_user( $wpdb->blogid, $current_user ) ) ) {
+		return;
+	}
+
+	remove_action( 'wp_head', array( 'Jetpack_Custom_CSS', 'link_tag' ), 101 );
+
+	if ( file_exists( ABSPATH . 'wp-content/plugins/jetpack/modules/private/private.php' ) ) {
```

Comment:
I would recommend against according the plugin path here, as this will cause issues on sites with a custom plugin directory location, or on sites where Jetpack is an mu-plugin. Instead you should be able to use `JETPACK__PLUGIN_DIR`